### PR TITLE
Update zlib entry; add instructions to get and use zlib-cloudflare

### PR DESCRIPTION
**Pull request by janaknat**
*Issue #, if available:*

*Description of changes:*
Provide instructions to get the src, build and install. Also, add info for AWS service users about zlib option.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,20 @@ bazel
...
+## zlib on Linux
+zlib-cloudflare has been updated to provide better and faster compression on Arm and x86. To use zlib-cloudflare:
+```
+git clone https://github.com/cloudflare/zlib.git
+cd zlib
+./configure
+make
+make install
+```
+For users of any supported AWS service, you can specify the library to be used for compression and decompression as:
```
Comment (AGSaidi) [inline README.md:103]: of Amazon Corretto ?

File: README.md [hunk 2]
```diff
@@ -91,6 +91,20 @@ bazel
 
 Bazelisk itself should not require further updates, as its only purpose is to keep Bazel updated.
 
+## zlib on Linux
```
Comment (AWSNB) [inline README.md:94]: does any of the OS distribution include this by default ?  if they do, then customer doesnt need to download and compiler.

actually, how can a developer check what zlib is running ? in fullness of time, if all OSes adopt cloudflare, then this step can be avoided

Comment (janaknat) [reply README.md:94]: I'll update with that info.

Comment (janaknat) [reply README.md:94]: @AWSNB I've added a step to determine which zlib is being used.

File: README.md [hunk 3]
```diff
@@ -91,6 +91,27 @@ bazel
...
+```
+If there are no results, then the original zlib is being used.
+
+zlib-cloudflare has been updated to provide better and faster compression on Arm and x86. To use zlib-cloudflare:
+```
+git clone https://github.com/cloudflare/zlib.git
+cd zlib
+./configure
+make
+make install
```
Comment (AGSaidi) [inline README.md:108]: here you're effectively saying override the system installed zlib package artifacts with this one manually. That isn't best practices.

Comment (janaknat) [reply README.md:108]: Updated to ./configure --prefix=$HOME. This will cause make install to install the file to $HOME.

File: README.md (hunk 2 above)
Comment (AGSaidi) [reply README.md:95]: None do today, this just takes up space. I'd just say, zlib packaged by distros doesn't include these optimizations. If you'd like to use them follow these steps and if that ever changes we update the doc.

**Pull request merged**